tests/test_web_api.py

```diff
@@ -0,0 +1,307 @@
+        be the same.
+        """
+        for key in expected:
+            self.assertEqual(expected[key], actual[key])
+        self.assertIn(actual["status"], job_status_values)
+
+    def test_create_ints(self):
+        data = {
+            "cfr_title": 0,
+            "cfr_part": 0,
+            "notification_email": "test@example.com"
+        }
+        response = self._postjson(data)
+
+        expected = {k: data[k] for k in data}
+        expected.update(self.defaults)
+        expected["url"] = status_url(fake_pipeline_id, sub_path="pipeline/")
+        self._stock_response_check(expected, response.data)
+        return expected
+
+    def test_create_strings(self):
+        data = {
+            "cfr_title": "0",
+            "cfr_part": "0",
+            "notification_email": "test@example.com"
+        }
+        response = self._postjson(data)
+
+        expected = {k: data[k] for k in data}
+        expected.update(self.defaults)
+        # Even if the input is a str, the return values should be ints:
+        expected["cfr_title"] = int(expected["cfr_title"])
+        expected["cfr_part"] = int(expected["cfr_part"])
+        expected["url"] = status_url(fake_pipeline_id, sub_path="pipeline/")
+        self._stock_response_check(expected, response.data)
+
+    def test_create_with_missing_fields(self):
+        data = {"cfr_part": "0"}
+        response = self._postjson(data)
+
+        self.assertEqual(400, response.status_code)
+        self.assertEqual({"cfr_title": ["This field is required."]},
+                         response.data)
+
+        data = {"cfr_title": "0"}
+        response = self._postjson(data)
+
+        self.assertEqual(400, response.status_code)
+        self.assertEqual({"cfr_part": ["This field is required."]},
+                         response.data)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(0, len(response.data))
+
+    def test_create_and_read(self):
+        expected = self.test_create_ints()
+
+        url = urlparse(expected["url"])
+        response = self.client.get(url.path, format="json")
+        self._stock_response_check(expected, response.data)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(1, len(response.data))
+        self._stock_response_check(expected, response.data[0])
+
+    def test_create_delete_and_read(self):
+        expected = self.test_create_ints()
+
+        url = urlparse(expected["url"])
+        response = self.client.delete(url.path, format="json")
+        self.assertEqual(204, response.status_code)
+
+        response = self.client.get(url.path, format="json")
+        self.assertEqual(404, response.status_code)
+
+        response = self.client.get("/rp/job/pipeline/", format="json")
+        self.assertEqual(0, len(response.data))
+
+
+class RegulationFileTestCase(APITestCase):
+
+    file_contents = "123"
+    hashed_contents = None
```

Comment:
Can we make this an instance attribute instead? Class-level variables don't play well with threading (which we'd use to run multiple tests in parallel)